rfcs/NNNN-standard-signals.md

```diff
@@ -0,0 +1,161 @@
+
+### Observing signal updates
+
+To enable observing signal changes, we need to run access to a signal in an _effect_ - a closure that contains the signal access and will be called again when the accessed signals change.
+
+Unlike most userland libraries, the standard signals proposal does not include an `effect()` helper (the intention is to leave effects and scheduling up to "frameworks"). Instead we must use a lower-level [_Watcher_ to watch signals and schedule updates](https://github.com/proposal-signals/proposal-signals/tree/main/packages/signal-polyfill).
+
+We can use a Watcher in two ways:
+1. To watch a computed signal that wraps the update lifecycle so that we're notified of updates to any signal read within the lifecycle and can trigger a new update.
+2. To watch individual signals bound directly to the DOM with a `watch()` directive to update just that binding.
+
+### SignalWatcher Mixin
+
+Conceptually, we want to run the reactive update lifecycle in an effect so that the signal library observes access to signals and trigger a new update.
+
+We can do this with an override of `performUpdate()` that wraps ReactiveElement's implementation in a watched computed signal:
+
+```ts
+  abstract class SignalWatcher extends Base {
+    private __watcher = new Signal.subtle.Watcher(() => {
+      this.requestUpdate();
+    });
+    private __updateSignal = new Signal.Computed(() => {
+      super.performUpdate();
+    });
+
+    override performUpdate() {
+      if (this.isUpdatePending === false) {
+        return;
+      }
+      this.__updateSignal.get();
+    }
+
+    override connectedCallback(): void {
+      this.__watcher.watch(this.__updateSignal);
+      super.connectedCallback();
+    }
+
+    override disconnectedCallback(): void {
+      this.__watcher.unwatch(this.__updateSignal);
+      super.disconnectedCallback();
+    }
+  }
+```
+
+### watch() directive
+
+The `watch()` async directive accepts a signal and renders its value _asynchronously_ to the containing binding. When the signal changes, the binding value is updated directly.
+
+Usage:
+```ts
+html`<p>${watch(messageSignal)}</p>`
+```
+
+We cannot write to the DOM synchronously like with `@lit-labs/preact-signals` because the standard signals proposal disallows reading signals from withing a watcher callback.
+
+Instead the updates must be asynchronous, but they can still be fine-grained. In cases where we have multiple signals updates and/or a element update in a microtask we will synchronize fine-grained updates and the element's update lifecycle.
+
+#### Static analysis of watch()
+
+`watch()` essentially unwraps a `Signal<T>`. This should be analyzable by template analyzers like lit-analyzer, but we need to check and ensure this is the case.
+
+### Auto-watching template tag
+
+Auto-watching versions of `html` and `svg` template tags will scan a template result's values and automatically wrap them in a `watch()` directive if they are signals.
```

Comment:
Can we add the ability to still pass a signal with the auto-watching template tag as a requirement?

e.g. `<my-component .signal=${signalRef(signal)} .signalValue=${signal}>`

or however it looks like

Reply:
Yeah, that's a good idea!